static/js/map/map.pokestop.js

```diff
@@ -25,7 +25,8 @@ function isPokestopMeetsQuestFilters(pokestop) {
                 return !settings.excludedQuestItems.has(id)
             }
             case 7: {
-                return !settings.excludedQuestPokemon.has(pokestop.quest.pokemon_id)
+                return !settings.excludedQuestPokemon.has(pokestop.quest.pokemon_id) &&
+                (settings.questFormFilter === 'Any' || settings.questFormFilter === getFormName(pokestop.quest.pokemon_id, pokestop.quest.form_id))
```

Comment:
`getFormName` returns the translated form name, hence why you need to also translate the select values.